File: Vanilla/sources/zomboid/zombie/world/moddata/ModData.java
package zombie.world.moddata;

import java.util.ArrayList;
import se.krka.kahlua.vm.KahluaTable;

public final class ModData {
   private static final ArrayList temp_list = new ArrayList();

   public static ArrayList getTableNames() {
      GlobalModData.instance.collectTableNames(temp_list);
      return temp_list;
   }

   public static boolean exists(String var0) {
      return GlobalModData.instance.exists(var0);
   }

   public static KahluaTable getOrCreate(String var0) {
      return GlobalModData.instance.getOrCreate(var0);
   }

   public static KahluaTable get(String var0) {
      return GlobalModData.instance.get(var0);
   }

   public static String create() {
      return GlobalModData.instance.create();
   }

   public static KahluaTable create(String var0) {
      return GlobalModData.instance.create(var0);
   }

   public static KahluaTable remove(String var0) {
      return GlobalModData.instance.remove(var0);
   }

   public static void add(String var0, KahluaTable var1) {
      GlobalModData.instance.add(var0, var1);
   }

   public static void transmit(String var0) {
      GlobalModData.instance.transmit(var0);
   }

   public static void request(String var0) {
      GlobalModData.instance.request(var0);
   }
}

File: Vanilla/sources/zdoc/media/lua/shared/Library/ModData.lua
---@class ModData : zombie.world.moddata.ModData
---@field private temp_list ArrayList|Unknown
ModData = {}

---@public
---@param arg0 String
---@return void
function ModData:request(arg0) end

---@public
---@return ArrayList|Unknown
function ModData:getTableNames() end

---@public
---@param arg0 String
---@return KahluaTable
function ModData:get(arg0) end

---@public
---@param arg0 String
---@param arg1 KahluaTable
---@return void
function ModData:add(arg0, arg1) end

---@public
---@param arg0 String
---@return KahluaTable
function ModData:remove(arg0) end

---@public
---@return String
---@overload fun(arg0:String)
function ModData:create() end

---@public
---@param arg0 String
---@return KahluaTable
function ModData:create(arg0) end

---@public
---@param arg0 String
---@return boolean
function ModData:exists(arg0) end

---@public
---@param arg0 String
---@return void
function ModData:transmit(arg0) end

---@public
---@param arg0 String
---@return KahluaTable
function ModData:getOrCreate(arg0) end
